File: JavaScript/5kyu/findUniq.js
/*
https://www.codewars.com/kata/585d8c8a28bc7403ea0000c3
DESCRIPTION: 
There is an array of strings. All strings contains similar 
letters except one. Try to find it!
    findUniq([ 'Aa', 'aaa', 'aaaaa', 'BbBb', 'Aaaa', 'AaAaAa', 'a' ]) === 'BbBb'
    findUniq([ 'abc', 'acb', 'bac', 'foo', 'bca', 'cab', 'cba' ]) === 'foo'
Strings may contain spaces. Spaces are not significant, 
only non-spaces symbols matters. E.g. string that contains 
only spaces is like empty string.
It’s guaranteed that array contains more than 2 strings.
*/
function findUniq(arr) {
    let newArr = arr.map(a => { 
        return [...new Set(a.toLowerCase())].sort().join("") 
    });
    
    for (let i = 0; i < newArr.length; i++) {
        if (newArr.indexOf(newArr[i]) === newArr.lastIndexOf(newArr[i])) { 
            return arr[i]; 
        } 
    } 
}

File: Java/6kyu/findUniq.java
// https://www.codewars.com/kata/585d7d5adb20cf33cb000235
// DESCRIPTION:
// There is an array with some numbers. All numbers are equal 
// except for one. Try to find it!
//     findUniq([ 1, 1, 1, 2, 1, 1 ]) === 2
//     findUniq([ 0, 0, 0.55, 0, 0 ]) === 0.55
// It’s guaranteed that array contains at least 3 numbers.
// The tests contain some very huge arrays, so think about 
// performance.

import java.util.Arrays; 

public class Kata { 

    public static double findUniq(double arr[]) { 
        Arrays.sort(arr);
        return arr[0] == arr[1] ? arr[arr.length-1]:arr[0];
    }
}
